x/application/types/genesis.go

```diff
@@ -21,14 +23,34 @@ func DefaultGenesis() *GenesisState {
 func (gs GenesisState) Validate() error {
 	// Check for duplicated index in application
 	applicationIndexMap := make(map[string]struct{})
-
 	for _, elem := range gs.ApplicationList {
 		index := string(ApplicationKey(elem.Address))
 		if _, ok := applicationIndexMap[index]; ok {
 			return fmt.Errorf("duplicated index for application")
 		}
 		applicationIndexMap[index] = struct{}{}
 	}
+
+	// Check that the stake value for the apps is valid
+	for _, elem := range gs.ApplicationList {
+		if elem.Stake == nil {
+			return fmt.Errorf("nil stake amount for application")
```

Comment:
Are these intentionally distinct from the errors defined in x/application/types/errors.go or should we be using those here as well?